actors/builtin/storage_miner/storage_miner_actor.go

```diff
@@ -31,6 +31,71 @@ const epochUndefined = abi.ChainEpoch(-1)
 
 type StorageMinerActor struct{}
 
+type CronEventType int64
+
+const (
+	CronEventType_Miner_SurpriseExpiration = CronEventType(1)
+	CronEventType_Miner_WorkerKeyChange    = CronEventType(2)
+	CronEventType_Miner_PreCommitExpiry    = CronEventType(3)
+	CronEventType_Miner_SectorExpiry       = CronEventType(4)
+	CronEventType_Miner_TempFault          = CronEventType(5)
+)
+
+type CronEventPayload struct {
+	EventType CronEventType
+	Sectors   []abi.SectorNumber // Empty for global events, such as SurprisePoSt expiration.
+}
+
+///////////////////////
+// Worker Key Change //
+///////////////////////
+
+func (a *StorageMinerActor) StageWorkerKeyChange(rt Runtime, newKey addr.Address) *vmr.EmptyReturn {
+	var st StorageMinerActorState
+	rt.State().Transaction(&st, func() interface{} {
+		rt.ValidateImmediateCallerIs(st.Info.Owner)
+
+		// must be BLS since the worker key will be used alongside a BLS-VRF
+		if newKey.Protocol() != addr.BLS {
+			rt.Abort(exitcode.ErrIllegalArgument, "Worker Key must be BLS.")
+		}
```

Comment:
Following on from my earlier comment, please leave a comment noting that this check isn't directly right, but we want to check that the account actor at the other end of this address has a BLS key.